From 2c68f0485c965447344a11026f7a1828433ec1d7 Mon Sep 17 00:00:00 2001
From: "Andrew F. Davis" <afd@ti.com>
Date: Fri, 18 Jan 2019 15:28:44 -0600
Subject: [PATCH 09/10] armV7R: K3: sysfw-loader: Move secure board config
 before UART access

The secure board config removes the default firewalls from the devices we
will use in the system. MAIN_UART0 is one such device that is by default
firewalled from us, send the secure config before we switch to that UART.

Signed-off-by: Andrew F. Davis <afd@ti.com>
---
 arch/arm/mach-k3/sysfw-loader.c | 38 +++++++++++++++++++-------------------
 1 file changed, 19 insertions(+), 19 deletions(-)

diff --git a/arch/arm/mach-k3/sysfw-loader.c b/arch/arm/mach-k3/sysfw-loader.c
index 063a858..a3d8d56 100644
--- a/arch/arm/mach-k3/sysfw-loader.c
+++ b/arch/arm/mach-k3/sysfw-loader.c
@@ -156,6 +156,25 @@ static void k3_sysfw_configure_using_fit(void *fit,
 		hang();
 	}
 
+	/* Extract security specific configuration from FIT */
+	ret = fit_get_data_by_name(fit, images, SYSFW_CFG_SEC,
+				   &cfg_fragment_addr, &cfg_fragment_size);
+	if (ret < 0) {
+		printf("Error accessing %s node in FIT (%d)\n", SYSFW_CFG_SEC,
+		       ret);
+		hang();
+	}
+
+	/* Apply security configuration to SYSFW */
+	ret = board_ops->board_config_security(ti_sci,
+					       (u64)(u32)cfg_fragment_addr,
+					       (u32)cfg_fragment_size);
+	if (ret) {
+		pr_err("Failed to set board security configuration (%d)\n",
+		       ret);
+		hang();
+	}
+
 	/*
 	 * Now that all clocks and PM aspects are setup, invoke a user-
 	 * provided callback function. Usually this callback would be used
@@ -181,25 +200,6 @@ static void k3_sysfw_configure_using_fit(void *fit,
 		pr_err("Failed to set board RM configuration (%d)\n", ret);
 		hang();
 	}
-
-	/* Extract security specific configuration from FIT */
-	ret = fit_get_data_by_name(fit, images, SYSFW_CFG_SEC,
-				   &cfg_fragment_addr, &cfg_fragment_size);
-	if (ret < 0) {
-		printf("Error accessing %s node in FIT (%d)\n", SYSFW_CFG_SEC,
-		       ret);
-		hang();
-	}
-
-	/* Apply security configuration to SYSFW */
-	ret = board_ops->board_config_security(ti_sci,
-					       (u64)(u32)cfg_fragment_addr,
-					       (u32)cfg_fragment_size);
-	if (ret) {
-		pr_err("Failed to set board security configuration (%d)\n",
-		       ret);
-		hang();
-	}
 }
 
 void k3_sysfw_loader(void (*config_pm_done_callback)(void))
-- 
2.7.4

